### DISCUSSION TITLE
2.1.0

### DISCUSSION THREAD

**Discussion by bgrainger** [mysql-net/MySqlConnector#1094] [General]
* Opening a connection from the pool is now twice as fast: #1089.
  * This may fail with Amazon Aurora RDS; to explicitly disable this, set `Pipelining = False;` in the connection string.
* Add support for [query attributes](https://dev.mysql.com/doc/refman/8.0/en/query-attributes.html): #930.
  * This requires MySQL Server 8.0.23 (or later) for regular commands, and MySQL Server 8.0.26 (or later) for prepared commands.
  * Add attributes to the `MySqlCommand.Attributes` collection to send them to the server when the command is executed.
* Add `MySqlDecimal` support: #1070.
* Improve `MySqlDataReader.GetX` exceptions for `NULL` values: #1092.
* Improve detection of Azure Database for MySQL proxies: #1093.
* Update Microsoft.SourceLink.GitHub.
* Thanks to @ejball and @sumitdvlp for contributions to this release.

## MySqlConnector.Logging.Microsoft.Extensions.Logging

* Logger names are now prefixed with `MySqlConnector.` by default: #1080.
  * To return to the previous behavior, use `MicrosoftExtensionsLoggingLoggerProvider(loggingFactory, omitMySqlConnectorPrefix: true)`.
* Added `.UseMySqlConnectorLogging()` extension method to add logging easily.

<hr /><em>This discussion was created from the release <a href='https://github.com/mysql-net/MySqlConnector/releases/tag/2.1.0'>2.1.0</a>.</em>

**Comment by MaceWindu** [comment]
I see that 2.1.x nugets ship assembly with 2.0.0 version. Is it expected change in v2 branch? v0/v1 releases had assembly version matching nuget version.

**Reply by bgrainger** [reply]
The version info on the DLL in the nupkg looks correct to me:

![image](https://user-images.githubusercontent.com/188129/154811034-9df16b8c-4cb0-424a-be45-93bf9e8e02e2.png)

Are you talking about the strong name? `MySqlConnector, Version=2.0.0.0, Culture=neutral, PublicKeyToken=d33d3e53aa5f8c92`

**Reply by bgrainger** [reply]
> Are you talking about the strong name?

If yes, then note that this is a recommended best practice:

> ✔️ CONSIDER only including a major version in the AssemblyVersion.

https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/versioning#assembly-version

**Reply by MaceWindu** [reply]
Yep, version in strong name (one, returned by `assembly.GetName().Version`). I'm fine with it, just wanted to confirm it's expected behavior, as we used version for feature detection for older versions.

**Reply by bgrainger** [reply]
OK, yes it is expected behaviour.

> v0/v1 releases had assembly version matching nuget version.

AFAIK they had a major-number-only strong name version, too. 

I don't actually know exactly where it comes from. The build chain generates `[assembly: System.Reflection.AssemblyVersionAttribute("2.0.0.0")]` in `src\MySqlConnector\obj\Release\net6.0\MySqlConnector.AssemblyInfo.cs` so something in the standard build chain decides to use only a major version.
